gridstatus/spp.py

```diff
@@ -1206,15 +1219,36 @@ def _format_5_min_url(
         # 2024-06-30 23:55:00 to 2024-07-01 00:00:00 is in the folder
         # 2024/06/By_Interval/30/
 
-        end = start + FiveMinOffset() if end is None else end.ceil("5min")
+        if end is None:
+            end = start + FiveMinOffset()
+        else:
+            # To deal with DST, convert to UTC before ceil
+            end = end.tz_convert("UTC").ceil("5min").tz_convert(self.default_timezone)
 
         folder_year = start.strftime("%Y")
         folder_month = start.strftime("%m")
         folder_day = start.strftime("%d")
 
         interval_str = "/By_Interval" if include_interval else ""
 
-        return f"{FILE_BROWSER_DOWNLOAD_URL}/{endpoint}?path=/{folder_year}/{folder_month}{interval_str}/{folder_day}/{file_prefix}-{end.strftime('%Y%m%d%H%M')}.csv"  # noqa
+        url = f"{FILE_BROWSER_DOWNLOAD_URL}/{endpoint}?path=/{folder_year}/{folder_month}{interval_str}/{folder_day}/{file_prefix}-{end.strftime('%Y%m%d%H%M')}.csv"  # noqa
+
+        status_code = requests.head(url).status_code
+
+        if status_code == 200:
```

Comment:
does this actually work as intended? here's the case im wondering about

the user requests `Nov 3, 2024 01:30am CST` . I think this function will first try to look for the file at `1:30am CDT`. if that file isn't found it will look for it during standard time version by appending the `d`. 

if the CDT file is found, it seems like this function will return the wrong url

i know this past year it looked like the CST files were missing, but i dont think that is guaranteed to be the case.

Reply:
You were correct. I've committed a fix that will work for both the CDT and CST files and checked it on last year when both sets of files were published.

https://github.com/gridstatus/gridstatus/pull/470/commits/f6ea436cafe063c47190792da76b71ec8f0587ba